Hangfire.Redis.Tests/RedisFetchedJobFacts.cs

```diff
@@ -90,6 +96,27 @@ public void RemoveFromQueue_RemovesOnlyJobWithTheSpecifiedId()
             });
         }
 
+        [Fact, CleanRedis]
+        public void RemoveFromQueue_DoesNotRemoveIfFetchedDoesntMatch()
+        {
+            UseRedis(redis =>
+            {
+                // Arrange
+				redis.ListRightPush("{hangfire}:queue:my-queue:dequeued", "job-id");
+
+                var fetchedAt = DateTime.UtcNow;
+                redis.HashSet("{hangfire}:job:job-id", "Fetched", JobHelper.SerializeDateTime(fetchedAt));
+                var fetchedJob = new RedisFetchedJob(_storage, redis, "job-id", "my-queue", fetchedAt + TimeSpan.FromSeconds(1));
+
+                // Act
+                fetchedJob.RemoveFromQueue();
+
+                // Assert
+                Assert.Equal(1, redis.ListLength("{hangfire}:queue:my-queue:dequeued"));
+                Assert.Equal("job-id", (string)redis.ListRightPop("{hangfire}:queue:my-queue:dequeued"));
+            });
+        }
+
         [Fact, CleanRedis]
         public void RemoveFromQueue_RemovesOnlyOneJob()
```

Comment:
I'm not really sure what this test is supposed to test. Having two jobs with the same `ID` in the `dequeued` list is probably a result of a bug. You can't save other dequeued job parameters in the job's hash. Completing the job will remove only one job from this list, but then what will happen to the other dequeued instance?

Reply:
Agree. I see that the test is part of the original test suite, It probably made sense at that time. There are no checks about the values in the dequeued list so it's possible that this list contains duplicate values, however, I agree, I believe it's just the result of a bug. Do you want to remove it ?

Reply:
Will do.